src/librustc/ty/layout.rs

```diff
@@ -606,34 +609,219 @@ impl<'a, 'tcx> LayoutCx<'tcx, TyCtxt<'a, 'tcx, 'tcx>> {
             }
 
             ty::Generator(def_id, ref substs, _) => {
-                // FIXME(tmandry): For fields that are repeated in multiple
-                // variants in the GeneratorLayout, we need code to ensure that
-                // the offset of these fields never change. Right now this is
-                // not an issue since every variant has every field, but once we
-                // optimize this we have to be more careful.
+                // When laying out generators, we divide our saved local fields
```

Comment:
This piece of code seems a bit long. Could it be split into functions?

Also I don't know too much about the layout code, so I'll ask @eddyb for a review of this.

Reply:
So nowadays, this code is in an `impl LayoutCx {...}`, which means some of the helper closures can be helper methods instead.

And this whole arm could be its own method. But it doesn't matter that much to me.